main.cpp

```diff
@@ -51,22 +51,322 @@ If you need inspiration for what to write, take a look at previously approved st
+{
+    cout << "DESTRUCTING Fighter" << endl;
+}
+
+string Fighter::fight(string weaponUsed)
+{
+    if (weaponUsed != "fists")
+    {
+        cout << "Fighter fights with " << weaponUsed << endl;
+        return "Fighter fights with " + weapon;
+    }
+    cout << "Fighter fights with fists" << endl;
+    return "Fighter fights with fists";
+}
+
+int Fighter::block(int speed)
+{
+    cout << "Fighter blocks with a speed of " << speed << endl;
+    return speed;
+}
+
+int Fighter::drink(int stamina)
+{
+    if (stamina > 12)
+    {
+        cout << "Fighter manages to drink another round!";
+        return 0;
+    }
+    cout << "Unable to hold his drink the Fighter collapses" << endl;
+    return 1;
+}
 /*
  UDT 3:
  */
+struct Preistess 
+{
+    Preistess();
+    ~Preistess();
+    string favoriteFood;
+    string leastFavoriteThing;
+    int crystals;
+    int illusions;
+    string home;
+    int castGrowSpell(int numPlants = 1);
+    string castBanish(string thingToBeBanished);
+    int castIllusions(int numIllusions = 1);
+};
+
+Preistess::Preistess() :
+favoriteFood("Salad"),
+leastFavoriteThing("Bad Smells"),
+crystals(12),
+illusions(20),
+home("Tree house")
+{
+    cout << "CONSTRUCTING Preistess with " << crystals << " ctrystals"  << endl;
+}
+
+Preistess::~Preistess()
+{
+    cout << "Deconstructing Preistess" << endl;
+}
+
+int Preistess::castGrowSpell(int numPlants)
+{
+    for (int i = 0; i < Preistess::crystals; i += 1)
+    {
+        numPlants += 1;
+    }
+    cout << "Priestess casts grow and " << numPlants << " plants grow immediately" << endl;
+    return numPlants;
+}
 
+string Preistess::castBanish(string thingToBeBanished)
+{
+    if (thingToBeBanished != Preistess::leastFavoriteThing)
+    {
+        cout << "Preistess casts banish and " << thingToBeBanished << " are banished for 2 rounds" << endl;
+        return "praretess casts banish and " + thingToBeBanished + " are banished for 2 rounds";
+    }
+    cout << "Preistess casts banish and " << Preistess::leastFavoriteThing << " are banished for 2 rounds" << endl;
+        return "preistess casts banish and " + Preistess::leastFavoriteThing + " are banished for 2 rounds";
+}
+
+int Preistess::castIllusions(int numIllusions)
+{
+    int index = Preistess::crystals;
+    while (index > 0)
+    {
+        numIllusions *= 2;
+        index --;
+    }
+    cout << "Preistess casts Illusions and " << numIllusions << " illusions appear instantly" << endl;
+    return numIllusions;
+}
 /*
  new UDT 4:
  */
+ struct Thief {
+     Thief();
+     ~Thief();
+     int tools;
+     string guild;
+     bool secretLanguage;
+     double speed = 33.50;
+     class Techniques {
+         Techniques(string main);
+         ~Techniques();
+         string main = "sneak";
+         string second = "climb";
+         string third = "back stab";
+         string fourth = "pick pocket";
+         string fifth = "read lips";
+     };
+     bool cartwheel(int dexterity = 13);
```

Comment:
`cartWheel` is a noun, not an action.
`performCartwheel` is more descriptive.

Name things what they are or what they do.
- [ ] revise